From: Richard B Winters <rik@mmogp.com>
Date: Mon, 4 May 2015 00:02:33 -0400
Subject: Fix check for systemctl on Debian

 - By default the Makefile checks for /usr/bin/systemctl in 
   order to determine if systemd is utilized by the 
   environment. On debian we can do this by checking pid 1

Change-Id: Ie4a490d5c2801c14c8b383bead370e93d7a4950c
Signed-off-by: Richard B Winters <rik@mmogp.com>
---
 Makefile | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/Makefile b/Makefile
index f318313..e0073e3 100644
--- a/Makefile
+++ b/Makefile
@@ -8,14 +8,7 @@ ifeq ($(PYTHON_SITELIB), /usr/lib/python2.7/dist-packages)
   EXTRA_SETUP_OPTS="--install-layout=deb"
 endif
 
-IS_SYSTEMCTL=false
-ifeq ("$(wildcard /usr/bin/systemctl)","/usr/bin/systemctl")
-  IS_SYSTEMCTL=true
-else
-  ifeq ("$(wildcard /bin/systemctl)","/usr/bin/systemctl")
-    IS_SYSTEMCTL=true
-  endif
-endif
+IS_SYSTEMCTL = $(shell if ps -p 1 -o comm= | grep -q -i "systemd" ; then echo true; else echo false; fi)
 
 MANDIR=/usr/share/man
 
@@ -47,9 +40,8 @@ install_pcsd:
 ifeq ($(IS_SYSTEMCTL),true)
 	install -d ${DESTDIR}/lib/systemd/system
 	install -m 644 pcsd/pcsd.service ${DESTDIR}/lib/systemd/system/
-else
-	install -m 755 -D pcsd/pcsd ${DESTDIR}/${initdir}/pcsd
 endif
+	install -m 755 -D pcsd/pcsd ${DESTDIR}/${initdir}/pcsd
 
 uninstall:
 	rm -f ${DESTDIR}${PREFIX}/sbin/pcs
@@ -57,9 +49,8 @@ uninstall:
 	rm -rf ${DESTDIR}${PREFIX}/share/pcsd
 ifeq ($(IS_SYSTEMCTL),true)
 	rm -f ${DESTDIR}/lib/systemd/system/pcsd.service
-else
-	rm -f ${DESTDIR}/${initdir}/pcsd
 endif
+	rm -f ${DESTDIR}/${initdir}/pcsd
 	rm -f ${DESTDIR}/etc/pam.d/pcsd
 	rm -rf ${DESTDIR}/var/lib/pcsd
 
